hoomd/dem/DEM2DForceCompute.h

```diff
@@ -38,7 +40,7 @@
   \ingroup computes
 */
 template<typename Real, typename Real4, typename Potential>
-class DEM2DForceCompute : public ForceCompute
+class DEM2DForceCompute : public ForceCompute, public DEMShapeSpecParser<Real, vec2<Real> >
```

Comment:
There must be a simpler solution than multiple inheritance. Why not just include
`getTypeShape` and `parseVertices` directly as members of the `DEM*ForceCompute`? No template specialization needed.

Also, `encodeVertices` would be a better function name. Parsing is converting a string into an object, encoding goes the other way.

Reply:
> There must be a simpler solution than multiple inheritance. Why not just include
> `getTypeShape` and `parseVertices` directly as members of the `DEM*ForceCompute`? No template specialization needed.
>
> Also, `encodeVertices` would be a better function name. Parsing is converting a string into an object, encoding goes the other way.

I can definitively include `getTypeShape` and `encodeVertices` as members of `DEM*ForceCompute` class and avoid the template specializations. However, I did the multiple inheritance to avoid duplicating the `getTypeShapeMapping` member method.

Reply:
Also included `getTypeShapeMapping` as a member of the DEM force computes in favor of avoiding multiple inheritance.